src/libs/antares/exception/LoadingError.cpp

```diff
@@ -41,8 +43,19 @@ InvalidSimulationMode::InvalidSimulationMode() :
 {
 }
 
-InvalidSolver::InvalidSolver(const std::string& solver) :
- LoadingError("Can't use solver '" + solver + "' : it does not match any available OR-Tools solver")
+static std::string InvalidSolverHelper(const std::string& solver, const std::list<std::string>& availableSolverList) {
+  std::ostringstream message;
+  message << "Can't use solver '" << solver << "' : it does not match any available OR-Tools solver. Possible choices are ";
+  for (const std::string& avail : availableSolverList)
+  {
+    bool last = &avail == &availableSolverList.back();
+    std::string sep = last ? "." : ", ";
+    message << std::quoted(avail) << sep;
+  }
```

Comment:
is it necessary to present available solver like this way?
if no:
 for (const std::string& solver :avail){
message <<solver<<"\n";}
if yes:
for (const std::string& solver :avail){
message <<solver<<", ";}
message.subtr....

Reply:
Yes, it's pretty much a user helping feature. It's nicer to present available options in case of failure.

Reply:
see this [available solvers](https://github.com/AntaresSimulatorTeam/Antares_Simulator/blob/5e462c03737157548caf48c1e039277621acb423/src/solver/misc/options.cpp#L56)

Reply:
Good point, see cc879ebba for correction & broader use of `availableOrToolsSolversString`, esp. in function `printSolvers`